p2p/crypto/ed25519.go

```diff
@@ -0,0 +1,110 @@
+package crypto
+
+import (
+	"errors"
+
+	ed "github.com/mildred/ed25519/src"
+)
+
+type edPubKey [32]byte
+type edPrivKey [64]byte
+type edSeed [32]byte
+
+type Ed25519PrivateKey struct {
+	sk ed.PrivateKey
+	pk ed.PublicKey
+}
+
+type Ed25519PublicKey struct {
+	k ed.PublicKey
+}
+
+func (pk *Ed25519PublicKey) Verify(data, sig []byte) (bool, error) {
+	if len(sig) != ed.SignatureSize {
+		return false, errors.New("Incorrect signature size")
+	}
+	var sig2 ed.Signature
+	copy(sig, sig2[:])
+	return ed.Verify(sig2, data, pk.k), nil
+}
+
+func (pk *Ed25519PublicKey) Bytes() ([]byte, error) {
```

Comment:
For the bytes method, its not well documented, but we want to use the protobuf structure in `p2p/crypto/internal/pb`. This allows us to easily detect and decode private and public keys (self describing data!)